csv_parser/test_case_generator.py

```diff
@@ -95,8 +161,49 @@ def run(perimeters):
     return test_cases
 
 
+def generate_step_json(perimeter, test_case, row, receive_json):
+    # Create the JSON file for the step
+    if not os.path.exists(f'./out/test-cases/{perimeter["name"]}/{test_case["label"]}'):
+        os.makedirs(f'./out/test-cases/{perimeter["name"]}/{test_case["label"]}')
+    with open(
+            f'./out/test-cases/{perimeter["name"]}/{test_case["label"]}/{len(test_case["steps"])}-{row["Pas de test"]} {row["Modèle"]}.json',
+            'w', encoding='utf-8') as file:
+        file.write(dumps(receive_json, indent=4, ensure_ascii=False))
+
+
 def get_type(type_in_french):
     if type_in_french == "Envoi":
         return "send"
     elif type_in_french == "Réception":
         return "receive"
+
+
+def generate_type_override_map(row):
+    # We load the relevant schema from ../src/main/resources/json-schema/row["Pas de test"].schema.json into the
+    # schema object
+    schema = json.load(open(f'../src/main/resources/json-schema/{row["Modèle"]}.schema.json'))
+    # We iterate over all properties recursively in the schem and add those that have simple types different from
```

Comment:
Tu itères sur les definitions là mais il faut effectivement aussi itérer sur les properties sinon tu pourrais planter sur un int à la racine par exemple (on en a pas a priori mais bon, on sait jamais)
cf exemple pour la différence entre les deux parties https://github.com/ansforge/SAMU-Hub-Modeles/blob/main/src/main/resources/json-schema/RS-EDA.schema.json